rust/datafusion/src/execution/context.rs

```diff
@@ -529,6 +525,15 @@ impl ExecutionConfig {
         self.query_planner = query_planner;
         self
     }
+
+    /// Adds a new [`OptimizerRule`]
+    pub fn add_optimizer_rule(
```

Comment:
I don't have a use case in mind, but what if we needed to insert an optimizer rule before the standard ones run, or override one of the standard rules?

Reply:
Had the same though too and also did not see an immediate use case. So here I just delayed the decision for that API to whenever/if someone has a need for it (`with_optimizer_rules` or something would work).